contrib/bindings/go/pathrs/pathrs.go

```diff
@@ -0,0 +1,328 @@
+	fd := C.pathrs_reopen(h.handle, C.int(flags))
+	err := handleErr(C.PATHRS_HANDLE, unsafe.Pointer(h.handle))
+	if err != nil {
+		return nil, err
+	}
+
+	name, err := randName(h.nameGen, 32)
+	if err != nil {
+		return nil, err
+	}
+
+	file := os.NewFile(uintptr(fd), "pathrs-handle:"+name)
+	return file, nil
+}
+
+// IntoFd unwraps a file-based libpathrs object to obtain its underlying file
+// descriptor.
+//
+// It is critical that you do not operate on this file descriptor yourself,
+// because the security properties of libpathrs depend on users doing all
+// relevant filesystem operations through libpathrs.
+func (h *Handle) IntoFd() (int, error) {
+	fd := int(C.pathrs_into_fd(C.PATHRS_HANDLE, unsafe.Pointer(h.handle)))
+	if fd < 0 {
+		return 0, handleErr(C.PATHRS_HANDLE, unsafe.Pointer(h.handle))
+	}
+
+	return fd, nil
+}
+
+// Clone creates a copy of root handler the new object will have a separate lifetime
+// from the original, but will refer to the same underlying file
+func (h *Handle) Clone() (*Handle, error) {
+	newHandler := (*C.pathrs_handle_t)(C.pathrs_duplicate(C.PATHRS_HANDLE, unsafe.Pointer(h.handle)))
+	err := handleErr(C.PATHRS_HANDLE, unsafe.Pointer(newHandler))
+	if err != nil {
+		return nil, err
+	}
+
+	return newHandle(newHandler), nil
+}
+
+// Free frees underling caught resources
+func (h *Handle) Free() {
+	if h == nil {
+		return
+	}
+
+	C.pathrs_free(C.PATHRS_HANDLE, unsafe.Pointer(h.handle))
+}
+
+// Error representation of rust error
+// particularly useful to not frighten to lost controll of pointer which can be rewritten.
+type Error struct {
+	description string
+	errno       uint64
+	backtrace   []backtraceLine
+}
+
+type backtraceLine struct {
+	ip       uintptr
+	sAddress uintptr
+	sName    string
+	sFile    string
+	sLineno  uint32
+}
+
+func (err *Error) Error() string {
+	return err.description
+}
+
+func (e *Error) Unwrap() error {
+	if e.errno != 0 {
+		return syscall.Errno(e.errno)
+	}
+
+	return nil
+}
+
+// Backtrace flush backtrace of underlying error to string.
+//
+// Its not passed to realization of Error interface on purpose since
+// the main error should remain clear and simple
+func (err *Error) Backtrace() string {
+	buf := strings.Builder{}
+
+	for _, line := range err.backtrace {
+		if line.sName != "" {
+			buf.WriteString(fmt.Sprintf("'%s'@", line.sName))
+		}
+		buf.WriteString(fmt.Sprintf("<0x%x>+0x%x\n", line.sAddress, line.ip-line.sAddress))
+		if line.sFile != "" {
+			buf.WriteString(fmt.Sprintf("  in file '%s':%d\n", line.sFile, line.sLineno))
+		}
+	}
+
+	return buf.String()
+}
+
+func newError(e *C.pathrs_error_t) error {
+	if e == nil {
+		return nil
+	}
+
+	err := &Error{
+		errno:       uint64(e.saved_errno),
+		description: C.GoString(e.description),
+		backtrace:   nil,
+	}
+
+	if e.backtrace != nil {
+		head := uintptr(unsafe.Pointer(e.backtrace.head))
+		length := uintptr(e.backtrace.length)
+		sizeof := unsafe.Sizeof(C.__pathrs_backtrace_entry_t{})
```

Comment:
It turns out you can do this with `C.sizeof___pathrs_backtrace_entry_t`. Then again, maybe we shouldn't...